library/common/main_interface.cc

```diff
@@ -258,21 +202,13 @@ envoy_status_t run_engine(envoy_engine_t, const char* config, const char* log_le
 
 void terminate_engine(envoy_engine_t) {
   // Reset the primary handle to the engine, but retain it long enough to synchronously terminate.
-  auto e = strong_engine_;
-  strong_engine_.reset();
+  auto e = EngineHandle::strong_engine_;
```

Comment:
instead of making these friend accesses to the EngineHandle class, why don't we add that API to the public EngineHandle class?

Reply:
My thinking was that we wanted to prevent consumers from being able to interfere with the init/termination flow, so by using friend functions we could limit access to only the main_interface.h functions. If we're okay with opening up access to init/start/terminate to the "public" (any additional C++ code that may pull this in) then I don't mind doing that and getting rid of all the friends

Reply:
Ok thanks for clarifying, that makes sense to me